main/http_server/axe-os/src/app/services/github-update.service.ts

```diff
@@ -1,32 +1,57 @@
 import { HttpClient } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
-import { map } from 'rxjs/operators';
-
+import { map, tap } from 'rxjs/operators';
+import { eASICModel } from 'src/models/enum/eASICModel';
 
 interface GithubRelease {
   id: number;
   tag_name: string;
   name: string;
+  body: string;
   prerelease: boolean;
 }
 
+interface ASICModelConfig {
+  supported_asic_models: string[];
+}
+
+interface ReleaseWithConfig {
+  release: GithubRelease;
+  supported_asic_models: string[];
+}
+
 @Injectable({
   providedIn: 'root'
 })
 export class GithubUpdateService {
+  constructor(private httpClient: HttpClient) {}
 
-  constructor(
-    private httpClient: HttpClient
-  ) { }
-
-
-  public getReleases(): Observable<GithubRelease[]> {
+  public getReleases(currentASICModel: eASICModel): Observable<GithubRelease[]> {
+    console.log('Fetching releases for ASIC model:', currentASICModel);
     return this.httpClient.get<GithubRelease[]>(
-      'https://api.github.com/repos/skot/esp-miner/releases'
+      'https://api.github.com/repos/Skot/ESP-Miner/releases'
     ).pipe(
-      map((releases: GithubRelease[]) => releases.filter((release: GithubRelease) => !release.prerelease))
+      tap(releases => console.log('Fetched releases:', releases)),
+      map(releases => releases.map(release => {
+        const supportedModels = this.parseSupportedModels(release.body);
```

Comment:
We should release an asset JSON file rather than relying on the description, that is very fragile.

https://api.github.com/repos/skot/ESP-Miner/releases/159514617/assets